distribution/bwc/build.gradle

```diff
@@ -197,21 +197,15 @@ bwcVersions.forPreviousUnreleased { VersionCollection.UnreleasedVersionInfo unre
         }
     }
 
-    if (gradle.startParameter.taskNames == ["assemble"]) {
-        // Gradle needs the `artifacts` declaration, including `builtBy` bellow to make projects dependencies on this
-        // project work, but it will also trigger the build of these for the `assemble` task.
-        // Since these are only used for testing, we don't want to assemble them if `assemble` is the single command being
-        // ran.
-        logger.info("Skipping BWC builds since `assemble` is the only task name provided on the command line")
-    } else {
-        artifacts {
-            for (File artifactFile : artifactFiles) {
-                String artifactName = artifactFile.name.contains('oss') ? 'elasticsearch-oss' : 'elasticsearch'
-                String suffix = artifactFile.toString()[-3..-1]
-                'default' file: artifactFile, name: artifactName, type: suffix, builtBy: buildBwcVersion
-            }
-        }
+    artifacts {
+      for (File artifactFile : artifactFiles) {
+        String artifactName = artifactFile.name.contains('oss') ? 'elasticsearch-oss' : 'elasticsearch'
+        String suffix = artifactFile.toString()[-3..-1]
+        'default' file: artifactFile, name: artifactName, type: suffix, builtBy: buildBwcVersion
+      }
     }
+    // make sure no dependencies were added to assemble; we want it to be a no-op
+    assemble.dependsOn = []
```

Comment:
It will (I tested it, and the passing CI shows this works too since it runs eg qa/full-cluster-restart which builds bwc). The dependent project triggers the builtBy task, not assemble.